Gekko/UI/AuthenticationViewController.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  AuthenticationViewController.swift
+//  Gekko
+//
+//  Created by Aleksandr Saliyenko on 3/13/18.
+//  Copyright © 2018 Sergii Mykhailov. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import LocalAuthentication
+
+class AuthenticationViewController : UIViewController {
+    
+    // MARK: Overriden methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let userDefaults = UserDefaults.standard
+        
+        publicKey = userDefaults.string(forKey:BTCTradeUAAccountSettingsViewController.PublicKeySettingsKey)
+        privateKey = userDefaults.string(forKey:BTCTradeUAAccountSettingsViewController.PrivateKeySettingsKey)
+        
+        self.setupMainIconImageView()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        if isAuthorized {
+            authenticationWithTouchID()
+        }
+        else {
+            performSegue(withIdentifier: AuthenticationViewController.ShowNavigationControllerSegueName, sender: self)
+        }
+    }
+    
+    // MARK: Internal methods
+    
+    fileprivate func setupMainIconImageView() {
+        let iconImage = (#imageLiteral(resourceName: "btcIcon"))
+        let mainIconImageView = UIImageView(image: iconImage)
+        
+        self.view.addSubview(mainIconImageView)
+        
+        mainIconImageView.layer.cornerRadius = mainIconImageView.frame.width / 2
+        mainIconImageView.layer.masksToBounds = true
+        
+        mainIconImageView.snp.makeConstraints { (make) in
+            let sizeOfMainView = self.view.frame.size
+            
+            make.centerX.equalTo(sizeOfMainView.width / 2)
+            make.centerY.equalTo(sizeOfMainView.height / 4)
+        }
+    }
+    
+    fileprivate var isAuthorized:Bool {
+        return publicKey != nil && !publicKey!.isEmpty && privateKey != nil && !privateKey!.isEmpty
+    }
+    
+    func authenticationWithTouchID() {
+        let localAuthenticationContext = LAContext()
+        localAuthenticationContext.localizedFallbackTitle = NSLocalizedString("Use Passcode", comment: "Fallback title")
+        
+        let reasonString = NSLocalizedString("To access the application", comment: "Authentication reason string")
+        
+        localAuthenticationContext.evaluatePolicy(.deviceOwnerAuthentication, localizedReason: reasonString) { (success, _) in
+            if success {
+                self.navigateToMainViewController()
+            }
+            else {
+                self.enableFailLabel()
+            }
+        }
+    }
+    
+    fileprivate func navigateToMainViewController() {
+        DispatchQueue.main.async { [weak self] in
```

Comment:
Seems that dispatch is not needed

Reply:
No I think it has to be here because this method is called from background